functions/Invoke-DbaLogShippingRecovery.ps1

```diff
@@ -0,0 +1,307 @@
+If this value is not supplied all databases will be recovered
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. To use:
+
+$scred = Get-Credential, then pass $scred object to the -SqlCredential parameter. 
+
+Windows Authentication will be used if SqlCredential is not specified. SQL Server does not accept Windows credentials being passed as credentials. 	
+To connect as a different Windows user, run PowerShell as that user.
+
+.PARAMETER NoRecovery
+Allows you to choose to not restore the database to a functional state (Normal) in the final steps of the process.
+By default the database is restored to a functional state (Normal). 
+
+
+.NOTES 
+Author: Sander Stad (@sqlstad), sqlstad.com
+Requires: sysadmin access on SQL Servers
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Invoke-DbaLogShippingRecovery
+
+.EXAMPLE   
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' 
+
+Recovers all the databases on the instance that are enabled for log shiping
+
+.EXAMPLE   
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' -SqlCredential $cred -Verbose
+
+Recovers all the databases on the instance that are enabled for log shiping using a credential
+
+.EXAMPLE   
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' -database 'db_logship' -Verbose
+
+Recovers the database "db_logship" to a normal status
+
+
+.EXAMPLE   
+db1, db2, db3, db4 | Invoke-DbaLogShippingRecovery -SqlServer 'server1' -Verbose
+
+Recovers the database db1, db2, db3, db4 to a normal status
+
+.EXAMPLE   
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' -Verbose -WhatIf
+
+Shows what would happen if the command were executed.
+
+#>
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+    param
+	(
+        [Parameter(Position=1, Mandatory=$true)]
+		[Alias("ServerInstance", "SqlInstance")]
+        [string]$SqlServer,
+        [Parameter(Position=2, Mandatory=$false, ValueFromPipeline=$true)]
+        [string[]]$Database = $null,
+        [Parameter(Position=3, Mandatory=$false)][object]$SqlCredential,
+        [Parameter(Position=4, Mandatory=$false)][switch]$NoRecovery
+	)
+
+    BEGIN
+    {
+        Write-Output "Attempting to connect to Sql Server.."
```

Comment:
Don't you worry, it's still sparkling new and nothing you must have known beforehand.
What Write-Message and Stop-Function do, is centralize our message and control flow.
Meaning, we have a central location that can ...
 - Determine how the user is informed about things
 - Handle logging / debugging information
This has the benefit, that we can place the choice easily into the hands of the user (using the no less new configuration system - think of that as a kind of environment-variables for dbatools in more shiny) on just how the system handles all that, without having to actually implement the consequence of that into each function individually.

For example, a user can configure, what level of message he wants written directly to screen, which he wants in Verbose and which in Warning. In case you wonder, the defaults are:
Write-Host/Write-Information: 1-3 (Automatically switches to Write-Information on v5)
Write-Verbose: 4-6
Write-Debug: 1-9 (So if you want to only write Debug, use something between 7-9)
The level system gives us the ability to control verbosity in an bit more granular fashion. That said, the level system does not affect _warnings_, which will always be shown unless the function is explicitly silenced.